app/Http/Controllers/SignupsController.php

```diff
@@ -121,7 +121,8 @@ public function show(Request $request, Signup $signup)
     {
         // Only allow an admin or the user who owns the signup to see the signup's unapproved posts.
         if ($request->query('include') === 'posts') {
-            $signup = Signup::withVisiblePosts()->first();
+            $signups = Signup::withVisiblePosts()->get();
```

Comment:
Are we grabbing all Signups and their visible posts here?

Reply:
@katiecrane yup! I wanted to still use this scope logic instead of recreating (thought it would get a little messy). Scopes return collections so we'll have to filter this more to grab the one signup we want.

Reply:
Hmmm does it not work to grab the Signup we want and then add `->withVisiblePosts()->get()`?

Reply:
@katiecrane that is what I thought would work too! But when trying that, it still returns a collection. From the [docs](https://laravel.com/docs/5.5/eloquent#local-scopes), it sounds like scopes query all Signups with certain constraints the scope defines. That is how I understand it but I could be wrong? cc @DFurnes @sbsmith86

Reply:
@chloealee when you do that it returns a collection of all posts or just of the post you initially grabbed?

Reply:
@katiecrane all signups with their visible posts!

Reply:
oh weirddd I would not have expected it to work that way!

Reply:
I know it took me a long time to figure that out! Again though, seems a little strange, so maybe I am wrong?

Reply:
@chloealee i think you are right about how scopes work. But, I don't think I understand the issue about the signup being returned as a collection? 

If you grab the signup: `Signup::find($signup_id)->withVisiblePosts()->get()`
You are saying you get a collection back, but that "collection" should only have one signup in it, right?

Reply:
@sbsmith86 I'm getting a collection back of all the signups :-/ 
![screen shot 2018-04-25 at 1 13 32 pm](https://user-images.githubusercontent.com/9019452/39261481-9861c59c-488a-11e8-91ad-48c05c01479d.png)

Reply:
This makes me pretty nervous - if this is loading _all_ signups and their visible posts, that's basically trying to fit all of Rogue's database into memory. Instead of using a query scope here, let's try making a [custom relationship](https://github.com/DoSomething/rogue/blob/0b5c135dc97ebbbef9f24bdc869a76e6125c67be/app/Models/Post.php#L90-L98) like we do for the `reaction` property on the Post.

Reply:
hmmmm also I am confused about the role check we are doing here: https://github.com/DoSomething/rogue/blob/22279d3705e67f53fca22693e360b756a6a12e3b/app/Models/Signup.php#L204-L209

when you call it as a staff user does this even do anything?

Reply:
Yeah, that also seems suspect.

Reply:
@DFurnes I thought about doing that too but then got confused how to add that to the query? Here, we want to only show the posts that the user should be able to see depending on their role but we don't necessarily want to grab those posts here, so I'm not sure we can do that? 

 @DFurnes @sbsmith86 yeah hmm looks like it shouldn't...

Reply:
oh @DFurnes @sbsmith86 I think the posts get included if you are staff through [this line](https://github.com/DoSomething/rogue/blob/master/app/Http/Controllers/SignupsController.php#L127) so we don't need to include there because staff/admins should be able to see everything?

Reply:
@chloealee You'd be operating on the provided `$signup` instance (rather than _all_ signups with `Signup`). If configured as a relationship, it should load as soon as you ask for the related model (e.g. when the transformer reads `$signup->visiblePosts`).

Reply:
We could also potentially do this in the `posts` relationship itself... and only apply the additional `where` queries if a user is logged in. Not sure if we'd run into any unexpected issues there, but it could be nice and clean!

Reply:
ah i like that idea, thanks for explaining @DFurnes !

Reply:
wait sorry - I am still a bit confused. If I did `$signup = $signup->visiblePosts;` (or updating the `posts` relation and grabbing that way), wouldn't it just return the signup's posts to me instead of the signup _with_ its posts?

Reply:
All of this is making me think it might be worth really planning to move towards policy classes to help with some of the "who can do what logic"

Reply:
@chloealee The relationships should be loaded in the transformer (where you linked above)! We want to keep `$signup` as the signup that we're trying to look at.

Reply:
@sbsmith86 agreed! Especially since Aaron just flagged another bug that admins can only delete signups, not staff members and there is another card for staff to upload photos in this sprint.

Reply:
@chloealee can you add a card for that so it is on our radar?

Reply:
@sbsmith86 for what Aaron mentioned or the policy work? [Here's](https://www.pivotaltracker.com/n/projects/2019429/stories/156700437) the card for the policy work! Checking with Jen right now about if I should tackle what Aaron mentioned along with the upload card that is set for this sprint or during bigger refactoring and creating policies.

Reply:
Ah great! never mind me :)

Reply:
> The relationships should be loaded in the transformer (where you linked above)! We want to keep $signup as the signup that we're trying to look at.

OH I see now @DFurnes! I didn't understand that part fully and was getting tripped up. Thanks!